### PR TITLE
Fix has_fit_methods

### PR DESCRIPTION
**Pull request by raphaelsaavedra**
Currently the `hasmethod` calls are not doing anything, the function always returns `true`

### CODE REVIEW

File: src/models/common.jl [hunk 1]
```diff
@@ -6,18 +6,18 @@ num_states(model::StateSpaceModel) = num_states(system(model))
...
-    hasmethod(initial_hyperparameters!, tuple_with_model_type)
-    hasmethod(constrain_hyperparameters!, tuple_with_model_type)
-    hasmethod(unconstrain_hyperparameters!, tuple_with_model_type)
-    hasmethod(update!, tuple_with_model_type)
-    return true
+    m1 = hasmethod(default_filter, tuple_with_model_type)
+    m2 = hasmethod(initial_hyperparameters!, tuple_with_model_type)
+    m3 = hasmethod(constrain_hyperparameters!, tuple_with_model_type)
+    m4 = hasmethod(unconstrain_hyperparameters!, tuple_with_model_type)
+    m5 = hasmethod(update!, tuple_with_model_type)
```
Comment (guilhermebodin) [inline src/models/common.jl:19]: `update!` was too vague as a function name. I changed it to:
`fill_model_system!(model::StateSpaceModel)` and
`fill_model_filter!(filter::KalmanFilter, model::StateSpaceModel)`

**Review by guilhermebodin — CHANGES_REQUESTED** [review]

File: src/models/common.jl (hunk 1 above)
Comment (raphaelsaavedra) [reply src/models/common.jl:19]: Only ARIMA and regression have `fill_model_filter!`, so I assume that's not a necessary fit method. I'll change `update!` to `fill_model_system!`

Comment (raphaelsaavedra) [reply src/models/common.jl:19]: By the way, some models have `update!` and others have `fill_model_filter!`. I assume this is an error? I'll change the remaining updates to `fill_model_filter!`. This should be failing some test, but apparently isn't.

Comment (guilhermebodin) [reply src/models/common.jl]: the models with `update!` are not working

**Review by guilhermebodin — APPROVED** [review]

File: src/models/common.jl (hunk 1 above)
Comment (raphaelsaavedra) [reply src/models/common.jl]: oh okay, then we should add that to the 0.4 milestones

**Pull request merged**